### DISCUSSION TITLE
Solidify the relationship between captains and their groups

### DISCUSSION THREAD

**Discussion by deammer** [distributeaid/shipment-tracker#533] [Questions]
We've made some assumptions about the relationship between group captains and their assigned groups throughout the code. It would be good to figure out an answer to the following questions:

**Can group captains be linked to multiple groups?**

-> If yes, we should update the code that assumes a one-to-one relationship between captains and groups
-> If not, we should update the code that allows a one-to-many relationship between captains and groups

### Context

This discussion sprouted from #132, where Bion wrote:

> The incongruity here is that in the backend there's a one-to-many association between user accounts and groups. I don't know if we need/want to support a person being a member of multiple groups. If we do it'd likely complicate the UI in addition to code you've added in this PR.
>
> The code as-written is mostly fine so long as we don't somehow end up with a user with multiple groups in the DB, which is possible at least as far as the backend is concerned.
>
> I'd be ok with more or less leaving it as is while we gather feedback and pay the tech debt in the future if need be to fully support one-to-many rather than just one-to-one. Short term, if we *don't* need to support one-to-many we should add validation checks in `src/resolvers/group.ts` to ensure it doesn't happen since right now I think the admin could update multiple groups to have the same captain..

_Originally posted by @bion in https://github.com/distributeaid/shipment-tracker/pull/132#discussion_r611370250_

**Comment by jtfairbank** [comment]
@deammer @bion two views here:

At any moment in time, I think a 1:1 association between Users & Groups is safe.  There are rare exceptions (Elliot), but he can just have different User accounts for different groups.  This may even be helpful as it makes it explicit which context he's logged into and which role he's acting in.  So that reflects a solid role-based view of User:Group relationships which is perfectly fine for our current efforts.  Relating to the tech world- this is kinda like having to sign up for each Slack that you're a part of.

Longer term, some volunteers tend to move around groups a fair bit, work on a different project in a different country but under the same group, change roles, etc.  There may also be cases where volunteers are temporarily lended across groups (such as a convoy of aid that includes volunteers and a 2 week stint at the receiving warehouse).  It'd be very cool to be able for volunteers to transition there account as this happens, in order to track their history & impact.  But that's a nice-to-have-in-the-future kind thing.  Relating this to the tech world- this is kinda like Github where you can participate in 1 or more project owned by 1 or more organization, but your profile tracks the overall history of your personal contributions.

Not exactly sure how that'd apply to the code here but I hope it helps clarify things and gives you some solid direction for now / later!
